src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,139 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (size == 0) {
+            head = newNode;
+        } else {
+            tail.nextElement = newNode;
+        }
+        tail = newNode;
+
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index > 0 && index < size) {
+            Node<T> atIndexNode = head;
+            for (int i = 0; i < index; i++) {
+                atIndexNode = atIndexNode.nextElement;
+            }
+            atIndexNode.prevElement.nextElement
+                    = new Node<>(atIndexNode.prevElement, value, atIndexNode);
+            size++;
+            return;
+        }
+        throw new IndexOutOfBoundsException(index + " Out of bounds");
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T t : list) {
+            add(t);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> currentNode = head;
+        if (index >= 0 && index < size) {
+            for (int i = 0; i < index; i++) {
+                currentNode = currentNode.nextElement;
+            }
+            return currentNode.element;
+        }
+        throw new IndexOutOfBoundsException(index + " Out of bounds");
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T result = get(index);
+        Node<T> atIndexNode = head;
+        if (isIndexExists(index)) {
+            for (int i = 0; i < index; i++) {
+                atIndexNode = atIndexNode.nextElement;
```

Comment:
you don't need this for loop, you just got your node by index, so you can just override it's value, Example: 
```
  Node<T> node = getNode(index);
        T oldValue = node.item;
        node.item = value;
        return oldValue;
```